fuel-vm/src/interpreter/executors/main/tests.rs

```diff
@@ -199,3 +199,51 @@ fn deploy__tx_with_wrong_gas_price_causes_error() {
         InterpreterError::ReadyTransactionWrongGasPrice { .. }
     ));
 }
+
+#[test]
+fn upgrade__tx_with_wrong_gas_price_causes_error() {
+    // given
+    let tx_gas_price = 1;
+    let interpreter_gas_price = 2;
+    let input_amount = 1000;
+    let arb_max_fee = input_amount;
+    let consensus_params = ConsensusParameters::standard();
+
+    let interpreter_params = InterpreterParams {
+        gas_price: interpreter_gas_price,
+        ..Default::default()
+    };
+
+    let ready_tx = TransactionBuilder::upgrade(UpgradePurpose::StateTransition {
```

Comment:
Similar to the "Upload" PR, this could be moved to a helper to declutter this test.